rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/RestConfiguration.java

```diff
@@ -72,14 +72,14 @@ public CurrentUserContext currentUserContext(TaskanaEngine taskanaEngine) {
 
   @Bean
   @ConditionalOnMissingBean(TaskanaEngine.class)
-  public TaskanaEngine getTaskanaEngine(TaskanaEngineConfiguration taskanaEngineConfiguration)
+  public TaskanaEngine getTaskanaEngine(SpringTaskanaEngineConfiguration taskanaEngineConfiguration)
       throws SQLException {
-    return taskanaEngineConfiguration.buildTaskanaEngine();
+    return SpringTaskanaEngine.buildTaskanaEngine(taskanaEngineConfiguration);
   }
 
   @Bean
-  @ConditionalOnMissingBean(TaskanaEngineConfiguration.class)
-  public TaskanaEngineConfiguration taskanaEngineConfiguration(DataSource dataSource) {
+  @ConditionalOnMissingBean(SpringTaskanaEngineConfiguration.class)
+  public SpringTaskanaEngineConfiguration taskanaEngineConfiguration(DataSource dataSource) {
```

Comment:
Do we really need `SpringTaskanaEngineConfiguration`? In my eyes we can remove that class.

Reply:
Maybe new subticket?